packages/core/src/addresses.rs

```diff
@@ -268,6 +266,9 @@ impl fmt::Display for Instantiate2AddressError {
     }
 }
 
+#[cfg(feature = "std")]
+impl std::error::Error for Instantiate2AddressError {}
```

Comment:
Isn't this something different than `#[derive(Error)]`? Not sure

Reply:
It depends, if we would provide a `#[source]` attribute we would need to overwrite the function. But if it is placed on an error with a `Display` impl, it will expand to essentially the same code (minus the `cfg` flag).

Most of the utility of `thiserror` for us came from it generating the `Display` implementations but `derive_more` can do the exact same.

When running `cargo expand` on this code:

```rust
#[derive(thiserror::Error)]
struct Thing;
```

it expands into:

```rust
/* --- snip prelude stuff --- */
struct Thing;
#[allow(unused_qualifications)]
impl std::error::Error for Thing { }
```